lib/jekyll/drops/document_drop.rb

```diff
@@ -9,8 +9,16 @@ class DocumentDrop < Drop
 
       def_delegator :@obj, :next_doc, :next
       def_delegator :@obj, :previous_doc, :previous
-      def_delegator :@obj, :relative_path, :path
       def_delegators :@obj, :id, :output, :content, :to_s, :relative_path, :url
+      def_delegators :@obj, :name, :name
+
+      def path
+        if @obj.is_a? Page
+          @obj.path
+        else
+          @obj.relative_path
+        end
+      end
```

Comment:
*Hacky code 2*: Pretty self-explanatory: `{{ page.path }}` should go to `path` for Pages, and `relative_path` for everything else. Weird thing is, `Page#path` is basically just the relative path, but overridable and without leading slashes.